contracts/MyToken.sol

```diff
@@ -1,17 +1,35 @@
 pragma solidity ^0.4.24;
 
 import "openzeppelin-solidity/contracts/token/ERC721/ERC721Token.sol";
+import "./DSToken.sol";
 
 contract MyToken is ERC721Token {
 
+  address daiAddress;
+
   constructor (string _name, string _symbol) public ERC721Token(_name, _symbol) {
+  }
+
 
+  function setDaiAddress(address _addr) public {
+    daiAddress = _addr;
   }
 
   function buyBook() public payable {
     require(msg.value == 1 ether);
     uint256 newTokenId = super.totalSupply() + 1;
     super._mint(msg.sender, newTokenId);
   }
+
+  function getDaiAddress() public returns (address) {
+    return daiAddress;
+  }
+
+  function buyBookWithDai() public payable {
+    DSToken dai = DSToken(daiAddress);
+    require(dai.transferFrom(msg.sender, address(this), 1));
```

Comment:
Is this basically hardcoding the price of the book to 1 DAI?

Reply:
This line seems to be doing several things:

1. invoking the transfer of 1 DAI
2. making sure the transfer went through

Maybe it should be broken up on two lines.

Reply:
Also, I think `address(this)` can be replaced with just `this`. See here: https://ethereum.stackexchange.com/questions/2806/how-do-you-refer-to-a-contract-own-address-inside-it

Reply:
Sure, I'll break this into 2 lines for readability. Plus changed to ```this```